src/hooks/useAuthFlow.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useContext } from 'react'
+import { ProviderType } from '@dcl/schemas'
+import { getConfiguration, connection } from 'decentraland-connect'
+import { FeatureFlagsContext, FeatureFlagsKeys } from '../components/FeatureFlagsProvider'
+import { fetchProfile } from '../modules/profile'
+import { locations } from '../shared/locations'
+import { isProfileComplete } from '../shared/profile'
+import { useNavigateWithSearchParams } from './navigation'
+import { useAfterLoginRedirection } from './redirection'
+import { useTargetConfig } from './targetConfig'
+
+export const useAuthFlow = () => {
+  const navigate = useNavigateWithSearchParams()
+  const { url: redirectTo } = useAfterLoginRedirection()
+  const { flags } = useContext(FeatureFlagsContext)
+  const [targetConfig] = useTargetConfig()
+
+  const shouldUseTestMagic = useCallback(() => {
+    return flags[FeatureFlagsKeys.MAGIC_TEST]
+  }, [flags])
```

Comment:
As `shouldUseTestMagic` returns `flags[FeatureFlagsKeys.MAGIC_TEST]`, what do you think about just using `flags[FeatureFlagsKeys.MAGIC_TEST]`? You can rename it into a different variable if you would like to, like `const shouldUseMagic = flags[FeatureFlagsKeys.MAGIC_TEST]`.